packages/vue-renderer/src/renderer.js

```diff
@@ -407,6 +413,46 @@ export default class VueRenderer {
     }
   }
 
+  async renderRoute(url, context = {}, retries = 5) {
+    /* istanbul ignore if */
+    if (!this.isReady) {
+      if (this.context.options.dev && retries > 0) {
+        consola.info('Waiting for server resources...')
+        await waitFor(1000)
+        return this.renderRoute(url, context, retries - 1)
+      } else {
+        throw new Error('Server resources are not available!')
+      }
+    }
+
+    // Log rendered url
+    consola.debug(`Rendering url ${url}`)
+
+    // Add url to the context
+    context.url = url
+
+    // context.spa
+    if (context.spa === undefined) {
+      // TODO: Remove reading from context.res in Nuxt3
```

Comment:
Why :thinking:

Reply:
Because module authors can still use `context.req` :)

Reply:
And better using the directly passed options + hooks

Reply:
Makes totally sense :speak_no_evil: